torch/nn/_functions/thnn/sparse.py

```diff
@@ -103,3 +103,152 @@ def backward(self, grad_output):
 
 
 _all_functions.append(Embedding)
+
+
+class EmbeddingBag(Function):
+
+    def __init__(self, max_norm, norm_type, scale_grad_by_freq, mode):
+        super(EmbeddingBag, self).__init__()
+        self.max_norm = max_norm
+        self.norm_type = norm_type
+        self.scale_grad_by_freq = scale_grad_by_freq
+        self._indices = None
+        assert mode is not None
+        if mode == 'sum':
+            self.mode = 0
+        elif mode == 'mean':
+            self.mode = 1
+        else:
+            raise InputError("mode needs to be 'sum' or 'mean', but got {}" % mode)
+
+    def _renorm(self, indices, weight):
+        self._backend.LookupTable_renorm(
+            self._backend.library_state,
+            indices,
+            weight,
+            self.max_norm,
+            self.norm_type
+        )
+
+    def forward(self, indices, offsets, weight):
+        assert indices.dim() <= 2
```

Comment:
Don't we only support batched inputs?
Also, use `if`s and raise proper errors instead of asserts